From fee5c1e4b789e41719af9fee0e2dd397cd31988f Mon Sep 17 00:00:00 2001
From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Fri, 3 Dec 2021 12:58:08 +0300
Subject: video: fbdev: atmel_lcdfb: fix an error code in atmel_lcdfb_probe()
Git-commit: fee5c1e4b789e41719af9fee0e2dd397cd31988f
Patch-mainline: v5.18-rc1
References: bsc#1129770

If "sinfo->config" is not found, then return -ENODEV.  Don't
return success.

Fixes: b985172b328a ("video: atmel_lcdfb: add device tree suport")
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Acked-by: Nicolas Ferre <nicolas.ferre@microchip.com>
Signed-off-by: Helge Deller <deller@gmx.de>
Acked-by: Thomas Zimmermann <tzimmermann@suse.de>
---
 drivers/video/fbdev/atmel_lcdfb.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/video/fbdev/atmel_lcdfb.c
+++ b/drivers/video/fbdev/atmel_lcdfb.c
@@ -1211,6 +1211,7 @@ static int __init atmel_lcdfb_probe(stru
 		goto free_info;
 	}
 
+	ret = -ENODEV;
 	if (!sinfo->config)
 		goto free_info;
 
